draw/test_transform.py, draw/test_visibility.py test fail on local Debian install

> Tests are known to pass for sure at least on Ubuntu (on CI), Fedora (for the packaging), Gentoo and Arch (on my computers). We can try on a fresh Debian installation in a container.
> 
> If you’re interested in finding a solution, could you please open a separate issue? It would be useful to share the failing results in `tests/draw/results` folder. 

 _Originally posted by @liZe in [#2553](https://github.com/Kozea/WeasyPrint/issues/2553#issuecomment-3718523762)_

On a fresh pull, 8 unit tests are failing. I am running

Debian testing
Python 3.13.7
GPL Ghostscript 10.06.0 (2025-09-09)
```
================================================ short test summary info =================================================
FAILED tests/draw/test_column.py::test_column_rule_normal - Failed: Pixel (0, 2) in column_rule_normal: expected rgba(255, 255, 255), got rgba(0, 0, 254)
FAILED tests/draw/test_transform.py::test_2d_transform_1 - Failed: Pixel (2, 1) in 2d_transform_1: expected rgba(255, 255, 255), got rgba(0, 0, 255)
FAILED tests/draw/test_transform.py::test_2d_transform_2 - Failed: Pixel (5, 1) in 2d_transform_2: expected rgba(255, 255, 255), got rgba(0, 0, 255)
FAILED tests/draw/test_transform.py::test_2d_transform_3 - Failed: Pixel (2, 4) in 2d_transform_3: expected rgba(255, 255, 255), got rgba(0, 0, 255)
FAILED tests/draw/test_transform.py::test_2d_transform_4 - Failed: Pixel (2, 4) in 2d_transform_4: expected rgba(255, 255, 255), got rgba(0, 0, 255)
FAILED tests/draw/test_visibility.py::test_visibility_1 - Failed: Pixel (6, 1) in visibility_1: expected rgba(255, 0, 0), got rgba(238, 0, 17)
FAILED tests/draw/test_visibility.py::test_visibility_3 - Failed: Pixel (6, 1) in visibility_3: expected rgba(255, 0, 0), got rgba(238, 0, 17)
FAILED tests/draw/test_visibility.py::test_visibility_4 - Failed: Pixel (6, 1) in visibility_4: expected rgba(255, 0, 0), got rgba(238, 0, 17)
====================================== 8 failed, 3783 passed, 39 xfailed in 55.79s =======================================
```
Images from `tests/draw/results/`:

<img width="8" height="8" alt="Image" src="https://github.com/user-attachments/assets/afb149b3-b6dc-433d-b937-01bbd85add7b" />
<img width="8" height="8" alt="Image" src="https://github.com/user-attachments/assets/a15c00dc-8695-4636-84c1-bc0de7b5e4e0" />
<img width="12" height="12" alt="Image" src="https://github.com/user-attachments/assets/9af6b7db-69a8-4bd8-8348-06f732de1a23" />
<img width="12" height="12" alt="Image" src="https://github.com/user-attachments/assets/5e876574-513e-43d8-b093-a0bfb7bf62a4" />
<img width="12" height="12" alt="Image" src="https://github.com/user-attachments/assets/fb1bf502-4775-493d-8661-229f71d1d22d" />
<img width="12" height="12" alt="Image" src="https://github.com/user-attachments/assets/40fc80d4-d4cf-423f-9690-059a6fc759dc" />
<img width="12" height="12" alt="Image" src="https://github.com/user-attachments/assets/df89e208-c6bf-4f6a-8099-f5e3f7bf73f2" />
<img width="12" height="12" alt="Image" src="https://github.com/user-attachments/assets/fe85d6f0-0130-4e68-9f46-5365006f3a86" />
<img width="16" height="8" alt="Image" src="https://github.com/user-attachments/assets/331c079d-48c0-40df-8a2c-fac7c93db275" />
<img width="16" height="8" alt="Image" src="https://github.com/user-attachments/assets/42a17d18-83d2-4c73-b11d-2e8d048083e9" />
<img width="5" height="5" alt="Image" src="https://github.com/user-attachments/assets/1e1a571d-0389-400b-813a-b7804d5c54b6" />
<img width="5" height="5" alt="Image" src="https://github.com/user-attachments/assets/4a7a58bb-a7ad-488f-b1d9-30fddd65aef6" />
<img width="5" height="5" alt="Image" src="https://github.com/user-attachments/assets/9eb9c722-f081-4603-bff6-dfadbead7ab6" />
<img width="5" height="5" alt="Image" src="https://github.com/user-attachments/assets/4e3cef07-20a0-4a86-80a4-bdf9bd12a09b" />
<img width="5" height="3" alt="Image" src="https://github.com/user-attachments/assets/0d8f9373-6d5b-47fe-b901-15d5e11371dd" />
<img width="5" height="3" alt="Image" src="https://github.com/user-attachments/assets/2c81eeea-4d90-4ab4-9690-60ea5ed98ade" />
<img width="2" height="2" alt="Image" src="https://github.com/user-attachments/assets/e8762d50-0141-4097-8c9c-0005f7f7a9a2" />
<img width="2" height="2" alt="Image" src="https://github.com/user-attachments/assets/4aa4864c-aaa3-4cc2-86d0-d4557384beda" />
<img width="10" height="10" alt="Image" src="https://github.com/user-attachments/assets/16171f25-7b7c-460b-a1a0-9596468dfa33" />
<img width="10" height="10" alt="Image" src="https://github.com/user-attachments/assets/ba1c4563-4e33-431b-985f-281917138996" />
<img width="16" height="10" alt="Image" src="https://github.com/user-attachments/assets/574a8d89-5934-4baf-96bd-0381f2d2d45f" />
<img width="16" height="10" alt="Image" src="https://github.com/user-attachments/assets/ea988ed4-0c9f-4abf-b610-78b0dc9f6e88" />
<img width="16" height="10" alt="Image" src="https://github.com/user-attachments/assets/0c0bd750-1e3d-4d2a-8305-224e11d9465b" />
<img width="16" height="10" alt="Image" src="https://github.com/user-attachments/assets/814db08e-af02-40d9-b825-3b233c26ed36" />
<img width="9" height="18" alt="Image" src="https://github.com/user-attachments/assets/c882c04a-c777-4eaa-a760-6b04a8ab11f0" />
<img width="9" height="18" alt="Image" src="https://github.com/user-attachments/assets/56259eee-78ee-4e2f-8132-6ffcc6d27941" />
<img width="4" height="4" alt="Image" src="https://github.com/user-attachments/assets/f82f6b58-aab8-499d-b4db-ae192d0e90b9" />
<img width="4" height="4" alt="Image" src="https://github.com/user-attachments/assets/32f39aee-c2f7-426c-a2e7-5ed7b119e049" />
<!-- Failed to upload "leader_bad_alignment.expected.png" -->
<!-- Failed to upload "leader_bad_alignment.png" -->
<img width="16" height="4" alt="Image" src="https://github.com/user-attachments/assets/a516db26-83cf-4f53-a352-a11c5ce0f8a6" />
<img width="16" height="4" alt="Image" src="https://github.com/user-attachments/assets/a8c4c241-0007-4f52-97d8-baa09baa574e" />
<img width="16" height="4" alt="Image" src="https://github.com/user-attachments/assets/3dbfa2eb-8f04-4645-a7f1-68157719039e" />
<img width="16" height="4" alt="Image" src="https://github.com/user-attachments/assets/bef72f5d-4e31-497c-bcee-339d73052ffd" />
<!-- Failed to upload "max_lines_nested.expected.png" -->
<img width="10" height="12" alt="Image" src="https://github.com/user-attachments/assets/bcffad11-08ed-4401-bc97-54c536f30d09" />
<img width="9" height="9" alt="Image" src="https://github.com/user-attachments/assets/a77d2744-d13d-42aa-a7da-0aa4afa60271" />
<img width="9" height="9" alt="Image" src="https://github.com/user-attachments/assets/3f08d93f-b8be-4780-aacd-0c3b6daf4e60" />
<img width="9" height="9" alt="Image" src="https://github.com/user-attachments/assets/585f6dfa-3c91-4327-a692-bda9bf0e60ed" />
<img width="28" height="52" alt="Image" src="https://github.com/user-attachments/assets/ecede1c3-1062-4f0c-9880-9b271b2a6900" />
<img width="28" height="52" alt="Image" src="https://github.com/user-attachments/assets/8acb4be2-0a1b-42d4-95f1-f6af65c15862" />
<!-- Failed to upload "tables_23.expected.png" -->
<img width="25" height="18" alt="Image" src="https://github.com/user-attachments/assets/4adb48b0-3b5f-41b1-b153-5cd4eac673ca" />
<!-- Failed to upload "text_float_text_decoration.expected.png" -->
<img width="13" height="7" alt="Image" src="https://github.com/user-attachments/assets/a28aaa52-5691-4ee1-bd70-37c10784818b" />
<!-- Failed to upload "text_nested_block_text_decoration.expected.png" -->
<img width="7" height="13" alt="Image" src="https://github.com/user-attachments/assets/9921ea83-7202-4185-b034-29a307195a99" />
<img width="13" height="7" alt="Image" src="https://github.com/user-attachments/assets/d112bf1c-f655-487b-a99a-43f0bee65459" />
<!-- Failed to upload "text_nested_text_decoration_color.png" -->
<!-- Failed to upload "visibility_1.expected.png" -->
<img width="12" height="7" alt="Image" src="https://github.com/user-attachments/assets/76728531-4bf4-429b-aca5-68f27fc5031f" />
<img width="12" height="7" alt="Image" src="https://github.com/user-attachments/assets/281fcc52-8ded-480c-a98a-5c03257189f5" />
<!-- Failed to upload "visibility_3.png" -->
<!-- Failed to upload "visibility_4.expected.png" -->
<!-- Failed to upload "visibility_4.png" -->

--- a/tests/draw/test_column.py
+++ b/tests/draw/test_column.py
@@ -93,7 +93,7 @@ def test_column_rule_normal(assert_pixels):
       <style>
         img { display: inline-block; width: 1px; height: 1px }
         div { columns: 2; column-gap: normal }
-        body { margin: 0; font-size: 3px; line-height: 0 }
+        body { margin: 0; font: 3px/0 weasyprint }
         @page { margin: 0; size: 5PX 3px }
       </style>
       <div>

--- a/tests/draw/test_transform.py
+++ b/tests/draw/test_transform.py
@@ -6,89 +6,86 @@ from ..testing_utils import assert_no_logs
 @assert_no_logs
 def test_2d_transform_1(assert_pixels):
     assert_pixels('''
-        ________
-        ________
-        __BBBr__
-        __BBBB__
-        __BBBB__
-        __BBBB__
-        ________
-        ________
+        __________
+        __________
+        __BBBr____
+        __BBBB____
+        __BBBB____
+        __BBBB____
+        __________
+        __________
+        __________
+        __________
     ''', '''
       <style>
-        @page { size: 8px; margin: 2px; }
-        div { transform: rotate(90deg); font-size: 0 }
+        @page { size: 10px; margin: 2px }
+        body { font-size: 0 }
+        img { transform: rotate(90deg) }
       </style>
-      <div><img src="pattern.png"></div>''')
+      <body><img src="pattern.png">''')
 
 
 @assert_no_logs
 def test_2d_transform_2(assert_pixels):
     assert_pixels('''
-        ____________
-        ____________
-        _____BBBr___
-        _____BBBB___
-        _____BBBB___
-        _____BBBB___
-        ____________
-        ____________
-        ____________
-        ____________
-        ____________
-        ____________
+        __________
+        __________
+        _____BBBr_
+        _____BBBB_
+        _____BBBB_
+        _____BBBB_
+        __________
+        __________
+        __________
+        __________
     ''', '''
       <style>
-        @page { size: 12px; margin: 2px; }
-        div { transform: translateX(3px) rotate(90deg);
-              font-size: 0; width: 4px }
+        @page { size: 10px; margin: 2px }
+        body { font-size: 0 }
+        img { transform: translateX(3px) rotate(90deg) }
       </style>
-      <div><img src="pattern.png"></div>''')
+      <body><img src="pattern.png">''')
 
 
 @assert_no_logs
 def test_2d_transform_3(assert_pixels):
     # A translateX after the rotation is actually a translateY
     assert_pixels('''
-        ____________
-        ____________
-        ____________
-        ____________
-        ____________
-        __BBBr______
-        __BBBB______
-        __BBBB______
-        __BBBB______
-        ____________
-        ____________
-        ____________
+        __________
+        __________
+        __________
+        __________
+        __________
+        __BBBr____
+        __BBBB____
+        __BBBB____
+        __BBBB____
+        __________
     ''', '''
       <style>
-        @page { size: 12px; margin: 2px; }
-        div { transform: rotate(90deg) translateX(3px);
-              font-size: 0; width: 4px }
+        @page { size: 10px; margin: 2px }
+        body { font-size: 0 }
+        img { transform: rotate(90deg) translateX(3px) }
       </style>
-      <div><img src="pattern.png"></div>''')
+      <body><img src="pattern.png">''')
 
 
 @assert_no_logs
 def test_2d_transform_4(assert_pixels):
     assert_pixels('''
-        ____________
-        ____________
-        ____________
-        ____________
-        ____________
-        __BBBr______
-        __BBBB______
-        __BBBB______
-        __BBBB______
-        ____________
-        ____________
-        ____________
+        __________
+        __________
+        __________
+        __________
+        __________
+        __BBBr____
+        __BBBB____
+        __BBBB____
+        __BBBB____
+        __________
     ''', '''
       <style>
-        @page { size: 12px; margin: 2px; }
+        @page { size: 10px; margin: 2px }
         div { transform: rotate(90deg); font-size: 0; width: 4px }
         img { transform: translateX(3px) }
       </style>

--- a/tests/draw/test_visibility.py
+++ b/tests/draw/test_visibility.py
@@ -5,8 +5,8 @@ from ..testing_utils import assert_no_logs
 visibility_source = '''
   <style>
     @page { size: 12px 7px }
-    body { font: 1px/1 serif }
-    img { margin: 1px 0 0 1px; }
+    body { font-size: 0; line-height: 0 }
+    img { margin: 1px 0 0 1px }
     %s
   </style>
   <div>

